lesson_09/test_homework_09.py

```diff
@@ -0,0 +1,98 @@
+import unittest
+from lesson_09.homework_functions_09 import MyFunctionsForTest
+
+class MyTests(unittest.TestCase):
+
+    def test_reverse_string_positive(self):
+        test_string = "Hello, world!"
+        expected_result = "!dlrow ,olleH"
+        self.assertEqual(MyFunctionsForTest.reverse_string(test_string), expected_result,
+                         f"Expected '{expected_result}' but got '{MyFunctionsForTest.reverse_string(test_string)}'")
+
+    def test_reverse_string_negative(self):
+        with self.assertRaises(TypeError, msg="Expected TypeError when passing a non-string value to reverse_string"):
+            MyFunctionsForTest.reverse_string(123456)
+
+
+
+    def test_longest_word_from_list_positive(self):
+        words = ["Kyiv", "Lviv", "Odesa", "Kharkiv", "Dnipro"]
+        expected_result = "Kharkiv"
+        self.assertEqual(MyFunctionsForTest.longest_word(words), expected_result,
+                         f"Expected '{expected_result}' but got '{MyFunctionsForTest.longest_word(words)}'")
+
+    def test_if_the_list_of_words_is_empty_positive(self):
+        words = []
+        expected_result = ""
+        self.assertEqual(MyFunctionsForTest.longest_word(words), expected_result,
+                         f"Expected '{expected_result}' but got '{MyFunctionsForTest.longest_word(words)}'")
+
+    def test_if_non_string_elements_negative(self):
+        words = ["Kyiv", 123, "Lviv", "Odesa", "Kharkiv", "Dnipro"]
+        with self.assertRaises(TypeError, msg="Expected TypeError when list contains non-string elements"):
+            MyFunctionsForTest.longest_word(words)
+
+
+
+    def test_mixed_list_positive(self):
+        input_list = ['1', '2', 3, True, 'False', 5, '6', 7, 8, 'Python', 9, 0, 'Lorem Ipsum']
+        result = MyFunctionsForTest.filter_string_from_list(input_list)
+        self.assertIn("1", result, "Expected '1' to be in the result list")
```

Comment:
For better optimise code - you can at least create for loop and iterate each object and assert is some data in and not in. Not to hard code each validation itself.